File: Source/AlphaTab/Rendering/Glyphs/BeatOnNoteGlyphBase.cs
using AlphaTab.Rendering.Utils;

namespace AlphaTab.Rendering.Glyphs
{
    internal class BeatOnNoteGlyphBase : BeatGlyphBase
    {
        public BeamingHelper BeamingHelper { get; set; }
        public float CenterX { get; set; }

        public BeatOnNoteGlyphBase()
        {
            CenterX = 0;
        }

        public virtual void UpdateBeamingHelper()
        {
        }

        //public override void Paint(float cx, float cy, ICanvas canvas)
        //{
        //    base.Paint(cx, cy, canvas);
        //    canvas.Color = Color.Random();
        //    canvas.FillRect(cx + X, cy + Y, Width, 100);
        //}
    }
}

File: src/rendering/glyphs/BeatOnNoteGlyphBase.ts
import { BeatGlyphBase } from '@src/rendering/glyphs/BeatGlyphBase';
import type { BeamingHelper } from '@src/rendering/utils/BeamingHelper';
import type { NoteXPosition, NoteYPosition } from '@src/rendering/BarRendererBase';
import type { Note } from '@src/model/Note';
import type { BeatBounds } from '@src/rendering/utils/BeatBounds';

export class BeatOnNoteGlyphBase extends BeatGlyphBase {
    public beamingHelper!: BeamingHelper;
    public centerX: number = 0;

    public updateBeamingHelper(): void {
        //
    }

    public buildBoundingsLookup(beatBounds: BeatBounds, cx: number, cy: number) {
        // implemented in subclasses
    }

    public getNoteX(note: Note, requestedPosition: NoteXPosition): number {
        return 0;
    }
    public getNoteY(note: Note, requestedPosition: NoteYPosition): number {
        return 0;
    }
}
